src/explorers/blockchain-explorer.js

```diff
@@ -0,0 +1,63 @@
+import { SUB_STEPS, TRANSACTION_APIS, TRANSACTION_ID_PLACEHOLDER } from '../constants';
+import { buildTransactionApiUrl } from '../services/transaction-apis';
+import { request } from '../services/request';
+import { VerifierError } from '../models';
+import { getText } from '../domain/i18n/useCases';
+import { generateTransactionDataFromBitpayResponse } from './bitcoin/bitpay';
+import { generateTransactionDataFromBlockcypherResponse } from './bitcoin/blockcypher';
+import { generateTransactionDataFromBlockexplorerResponse } from './bitcoin/blockexplorer';
+import { generateTransactionDataFromBlockstreamResponse } from './bitcoin/blockstream';
+import { generateTransactionDataFromEtherscanResponse } from './ethereum/etherscan';
+import { isTestChain } from '../constants/blockchains';
+import { generateTransactionDataFromBlockcypherEthereumResponse } from './ethereum/blockcypher-ethereum';
+
+function isTransactionGeneratorAsync (chain) {
+  return chain === TRANSACTION_APIS.Etherscan;
+}
+
+export async function getBlockchainTransactionFromApi (apiName, transactionId, chain) {
+  const requestUrl = buildTransactionApiUrl({
+    apiName,
+    searchValue: TRANSACTION_ID_PLACEHOLDER,
+    newValue: transactionId,
+    testApi: isTestChain(chain)
+  });
+
+  return new Promise((resolve, reject) => {
+    return request({ url: requestUrl }).then(response => {
+      try {
+        const txMethod = getTransactionDataGeneratorPerApi(apiName);
+        if (isTransactionGeneratorAsync(chain)) {
+          txMethod(JSON.parse(response), chain).then((transactionData) => {
+            resolve(transactionData);
+          }).catch(err => {
+            reject(err.message);
+          });
+        } else {
+          resolve(txMethod(JSON.parse(response), chain));
+        }
+      } catch (err) {
+        reject(err.message);
+      }
```

Comment:
Honestly have no clue what the transaction generator is supposed to do and why it's different from the rest. End result looks pretty close to the same either way.

Reply:
The transaction generator is just there to process the transaction API request and create a TransactionData. The goal is for it to be used by all APIs. 
I didn't write the code to get a transaction from Ethscan, but the reason why it's async is that it does 2 more API requests after getting the original transaction. I'm not experienced enough to decide whether or not it can be simplified, at least not without spending some solid time learning more about blockchain.

Reply:
I can sympathize with that, because I do know some API's are more problematic than most. In eth land, most use and/or mimic GETH API's so sometimes it just be like that. 

However the more that can be abstracted out, the better. I can point you to some python code that does a decent job at this if you need some pointers. We hit some of the same API's and more.